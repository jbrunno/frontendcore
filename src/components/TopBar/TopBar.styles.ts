import { css, styled } from '@mui/material/styles';
import AppBar from '../AppBar/AppBar';
import Card from '../Card/Card';
import Menu from '../Menu/Menu';

export const AppBarStyled = styled(AppBar)(
  ({ theme }) => css`
    border-radius: 0 0 ${theme.fn.size(16)} ${theme.fn.size(16)};
    background-color: ${theme.palette.common.white};
  `,
);

export const MenuStyled = styled(Menu)`
  left: ${({ theme }) => theme.fn.spacing(-105)};
`;

export const DialogCardStyled = styled(Card)(
  ({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.fn.spacing(20)};
    padding: ${theme.fn.spacing(22, 32, 32, 32)};
    width: ${theme.fn.size(350)};
  `,
);
