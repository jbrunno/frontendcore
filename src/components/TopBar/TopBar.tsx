import {
  CarteiraUserAccount,
  Children,
  ContextoUserAccount,
  ObjectLiteral,
} from '@/types';
import { MoreVert } from '@mui/icons-material';
import { AutocompleteRenderInputParams } from '@mui/material';
import Autocomplete from '../Autocomplete/Autocomplete';
import Button from '../Button/Button';
import Dialog from '../Dialog/Dialog';
import { Grid } from '../Grid/Grid';
import IconButton from '../IconButton/IconButton';
import MenuItem from '../MenuItem/MenuItem';
import { SearchField, SearchFieldProps } from '../SearchField/SearchField';
import TextField from '../TextField/TextField';
import { Typography } from '../Typography/Typography';
import { useTopBar } from './hooks/useTopBar';
import { AppBarStyled, DialogCardStyled, MenuStyled } from './TopBar.styles';
import { AreaNomeEnum } from '../../types/data';
import Tooltip from '../Tooltip/Tooltip';

export type TopBarProps = {
  children?: Children;
};

export function TopBar({ children, ...props }: TopBarProps & SearchFieldProps) {
  const {
    anchorEl,
    openMenu,
    openAreasDialog,
    openCarteirasDialog,
    areas,
    carteiras,
    currentArea,
    currentCarteira,
    selectArea,
    selectCarteira,
    handleAreaChange,
    handleCarteiraChange,
    toggleMenuOpen,
    toggleAreasDialog,
    toggleCarteirasDialog,
  } = useTopBar();

  return (
    <AppBarStyled elevation={0} position="sticky" color="default">
      <Grid
        container
        alignItems="center"
        justifyContent="space-between"
        paddingY={2}
        paddingX={4}
      >
        <Grid item xs={5}>
          <SearchField
            size="small"
            placeholder="Buscar por CPF, título ou nome"
            fullWidth
            {...props}
          />
        </Grid>
        {children || (
          <Grid
            item
            xs={4}
            container
            alignItems="center"
            justifyContent="right"
          >
            {currentArea && (
              <Typography variant="body1" color="textSecondary">
                {AreaNomeEnum[currentArea.nome as keyof typeof AreaNomeEnum]}
              </Typography>
            )}
            {currentCarteira && (
              <Typography variant="body1" color="textSecondary">
                &nbsp;-&nbsp;
              </Typography>
            )}
            {currentCarteira && (
              <Typography variant="subtitle1" color="textPrimary">
                {currentCarteira.nome}
              </Typography>
            )}
            {(areas || carteiras) && (
              <Tooltip title="Em desenvolvimento">
                <IconButton color="inherit" onClick={toggleMenuOpen}>
                  <MoreVert fontSize="small" />
                </IconButton>
              </Tooltip>
            )}
          </Grid>
        )}
      </Grid>
      <MenuStyled
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={openMenu}
        onClose={toggleMenuOpen}
      >
        <MenuItem onClick={toggleAreasDialog}>Alterar área</MenuItem>
        <MenuItem onClick={toggleCarteirasDialog}>Alterar carteira</MenuItem>
      </MenuStyled>
      <Dialog open={openAreasDialog} onClose={toggleAreasDialog}>
        <DialogCardStyled>
          <Typography mb={1} variant="h3" color="grey.800">
            Alterar área de atuação
          </Typography>
          <Autocomplete
            fullWidth
            size="small"
            defaultValue={currentArea}
            renderInput={(params: AutocompleteRenderInputParams) => (
              <TextField
                {...params}
                variant="outlined"
                label="Selecione a área de atuação"
              />
            )}
            getOptionLabel={(option) =>
              AreaNomeEnum[
                (option as ObjectLiteral<string, string>)
                  .nome as keyof typeof AreaNomeEnum
              ]
            }
            onChange={(_, value) => selectArea(value as ContextoUserAccount)}
            options={areas || []}
          />
          <Button fullWidth variant="contained" onClick={handleAreaChange}>
            Alterar
          </Button>
        </DialogCardStyled>
      </Dialog>
      <Dialog open={openCarteirasDialog} onClose={toggleCarteirasDialog}>
        <DialogCardStyled>
          <Typography mb={1} variant="h3" color="grey.800">
            Alterar carteira
          </Typography>
          <Autocomplete
            fullWidth
            size="small"
            defaultValue={currentCarteira}
            renderInput={(params: AutocompleteRenderInputParams) => (
              <TextField
                {...params}
                variant="outlined"
                label="Selecione a carteira"
              />
            )}
            getOptionLabel={(option) =>
              (option as ObjectLiteral<string, string>).nome
            }
            onChange={(_, value) =>
              selectCarteira(value as CarteiraUserAccount)
            }
            options={carteiras || []}
          />
          <Button fullWidth variant="contained" onClick={handleCarteiraChange}>
            Alterar
          </Button>
        </DialogCardStyled>
      </Dialog>
    </AppBarStyled>
  );
}

export default TopBar;
