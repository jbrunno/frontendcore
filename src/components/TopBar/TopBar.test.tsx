import { render } from '@test-utils/component-testing';
import { TopBar } from './TopBar';

test('renders the TopBar', () => {
  const { container } = render(<TopBar />);
  expect(container).toMatchSnapshot();
});
