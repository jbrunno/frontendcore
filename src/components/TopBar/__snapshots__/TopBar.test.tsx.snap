// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TopBar 1`] = `
.emotion-1 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  background-color: #f5f5f5;
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0 0 calc(0.7rem + 4.8px) calc(0.7rem + 4.8px);
  background-color: #fff;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: 1rem;
  padding-bottom: 1rem;
  padding-left: 2rem;
  padding-right: 2rem;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 41.666667%;
  -ms-flex-preferred-size: 41.666667%;
  flex-basis: 41.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 41.666667%;
}

@media (min-width:600px) {
  .emotion-3 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:900px) {
  .emotion-3 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:1200px) {
  .emotion-3 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:1536px) {
  .emotion-3 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-5 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
  padding-left: 14px;
  padding-right: 14px;
}

.emotion-5.Mui-disabled {
  color: #3B3B4161;
  cursor: default;
}

.emotion-5:hover .MuiOutlinedInput-notchedOutline {
  border-color: #3B3B41DE;
}

@media (hover: none) {
  .emotion-5:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-5.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #4A31FF;
  border-width: 2px;
}

.emotion-5.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #F44336;
}

.emotion-5.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: #00000042;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: #0000008A;
  margin-right: 8px;
}

.emotion-7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-8 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
  padding-left: 0;
  padding-right: 0;
}

.emotion-8::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8:focus {
  outline: 0;
}

.emotion-8:invalid {
  box-shadow: none;
}

.emotion-8::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-8:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-8.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #3B3B4161;
}

.emotion-8:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-8:-webkit-autofill {
  border-radius: inherit;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: #0000008A;
  margin-left: 8px;
}

.emotion-10 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B4199;
}

.emotion-11 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-12 {
  float: unset;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-13 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 33.333333%;
  -ms-flex-preferred-size: 33.333333%;
  flex-basis: 33.333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 33.333333%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

@media (min-width:600px) {
  .emotion-13 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:900px) {
  .emotion-13 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1200px) {
  .emotion-13 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1536px) {
  .emotion-13 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionSticky emotion-0 emotion-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container emotion-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 emotion-3"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-4"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart MuiInputBase-adornedEnd emotion-5"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall emotion-6"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-7"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd emotion-8"
              id=":r0:"
              placeholder="Buscar por CPF, título ou nome"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall emotion-9"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-10"
              />
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline emotion-11"
            >
              <legend
                class="emotion-12"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4 emotion-13"
      />
    </div>
  </header>
</div>
`;
